### PR TITLE
compute: wallclock lag summary metric

### PR DESCRIPTION
**Pull request by teskje**
This PR improves on the existing dataflow wallclock lag metric by making three changes:

 * Drops the `avg` aggregation in favor of total sum and total count.
 * Adjusts the metric names to immitate the summary metric type. We cannot use an actual summary metric because the `prometheus` crate doesn't support those.
 * Changes the aggregation approach to exposing the min/max value over the sliding window of the most recent 60 samples, instead of updating the metric values once per minute.

For the last part, a `MinMax` datastructure is introduced to efficiently keep track of the min/max values are the sample window slides. Some care is taken to reduce the memory usage of that datastructure to roughly 750 bytes per compute collection.

### Motivation

  * This PR adds a known-desirable feature.

Part of #28097 

See discussion in #29140 

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/compute-client/src/metrics.rs [hunk 1]
```diff
@@ -454,70 +473,157 @@ impl StatsCollector<ProtoComputeCommand, ProtoComputeResponse> for ReplicaMetric
...
+    /// +----------------^---------------^
+    ///           push_stack_len
+    ///
+    /// We keep both the push stack and the pop stack in the same `Vec`, to guarantee that we only
+    /// allocate `size_of::<T>() * 3 * window_size` bytes. If we kept two `Vec`s we'd need twice
+    /// that much.
+    stacks: Vec<(T, T, T)>,
+    /// The length of the push stack.
+    push_stack_len: usize,
 }
```
Comment (benesch) [inline src/compute-client/src/metrics.rs:533]: This is so so cool. I had to stare at it for a while to get an intuition for how it works, but I get it now. You fill up the whole vector up to its capacity with the push stack. Then when you hit the capacity, you flip it. Then you fill it up again one at a time, where each new entry on the push stack overwrites the previous entry on the pop stack. And when you get to the end, you flip it again, and start over. Genius!

File: src/compute-client/src/metrics.rs [hunk 2]
```diff
@@ -704,3 +810,34 @@ impl<M> PeekMetrics<M> {
...
+        push_and_check(9, (1, 10));
+        push_and_check(3, (1, 10));
+        push_and_check(8, (2, 10));
+        push_and_check(5, (2, 9));
+        push_and_check(5, (3, 9));
+        push_and_check(5, (3, 8));
+        push_and_check(5, (5, 8));
+        push_and_check(5, (5, 5));
+    }
+}
```
Comment (benesch) [inline src/compute-client/src/metrics.rs:843]: This feels general purpose enough that I think it should move to the `ore::stats` module. No need to do taht in this PR though.

**Review by benesch — APPROVED** [review]
Absolutely brilliant, @teskje! Thanks for turning this into reality overnight!

File: src/compute-client/src/metrics.rs (hunk 1 above)
Comment (teskje) [reply src/compute-client/src/metrics.rs:533]: Indeed! I'll explain this better when moving this to ore.

**Comment by teskje** [comment]
TFTR!

**Pull request merged**